common/scala/src/main/scala/whisk/common/TransactionId.scala

```diff
@@ -27,33 +27,95 @@ import spray.json.JsArray
 import spray.json.JsNumber
 import spray.json.JsValue
 import spray.json.RootJsonFormat
+import java.time.Duration
+import akka.event.Logging.LogLevel
+import akka.event.Logging.{ InfoLevel, WarningLevel }
 
 /**
  * A transaction id for tracking operations in the system that are specific to a request.
- * An instance of TransactionId is implicitly received by all logging methods. This is a
- * value type so that the type system ensures it is never null. The actual metadata is stored
- * indirectly in the referenced meta object.
+ * An instance of TransactionId is implicitly received by all logging methods. The actual
+ * metadata is stored indirectly in the referenced meta object.
  */
-case class TransactionId private (meta: TransactionMetadata) extends AnyVal {
+case class TransactionId private (meta: TransactionMetadata) extends Logging {
     def apply() = meta.id
     override def toString = if (meta.id > 0) s"#tid_${meta.id}" else (if (meta.id < 0) s"#sid_${-meta.id}" else "??")
 
     /**
-     * Computes elapsed milliseconds since start of transaction if given token is defined.
+     * Method to count events.
      *
-     * @param token the marker name
-     * @return Some LogMarker iff token is defined and None otherwise
+     * @param from Reference, where the method was called from.
+     * @param marker A LogMarkerToken. They are defined in <code>LoggingMarkers</code>.
+     * @param message An additional message that is written into the log, together with the other information.
+     * @param logLevel The Loglevel, the message should have. Default is <code>InfoLevel</code>.
      */
-    def mark(token: LogMarkerToken): Option[LogMarker] = {
-        val now = Instant.now(Clock.systemUTC())
-        Option(token) filter { _.toString.trim.nonEmpty } map { _ =>
-            val delta = now.toEpochMilli - meta.start.toEpochMilli
-            LogMarker(now, delta, token.toString.trim)
-        }
+    def counting(from: AnyRef, marker: LogMarkerToken, message: String = "", logLevel: LogLevel = InfoLevel) = {
+        emit(logLevel, this, from, message, Some(LogMarker(marker, deltaToStart)))
     }
+
+    /**
+     * Method to start taking time of an action in the code. It returns a <code>StartMarker</code> which has to be
+     * passed into the <code>finished</code>-method.
+     *
+     * @param from Reference, where the method was called from.
+     * @param marker A LogMarkerToken. They are defined in <code>LoggingMarkers</code>.
+     * @param message An additional message that is written into the log, together with the other information.
+     * @param logLevel The Loglevel, the message should have. Default is <code>InfoLevel</code>.
+     *
+     * @return startMarker that has to be passed to the finished or failed method to calculate the time difference.
+     */
+    def starting(from: AnyRef, marker: LogMarkerToken, message: String = "", logLevel: LogLevel = InfoLevel): StartMarker = {
```

Comment:
`started` for symmetry with `finished` (i concede that will precede the actual "starting").